packages/ifpack2/src/Ifpack2_ILUT_def.hpp

```diff
@@ -177,27 +177,28 @@ void ILUT<MatrixType>::setParameters (const Teuchos::ParameterList& params)
   if (! gotFillLevel) {
     // Try magnitude_type, for backwards compatibility.
     try {
-      fillLevel = as<int> (params.get<magnitude_type> ("fact: ilut level-of-fill"));
+      fillLevel = static_cast<double> (params.get<magnitude_type> ("fact: ilut level-of-fill"));
     }
     catch (InvalidParameterType&) {}
   }
   if (! gotFillLevel) {
-    // Try double, for backwards compatibility.
+    // Try int, for backwards compatibility.
     try {
-      fillLevel = as<int> (params.get<double> ("fact: ilut level-of-fill"));
+      fillLevel = static_cast<double> (params.get<int> ("fact: ilut level-of-fill"));
     }
     catch (InvalidParameterType&) {}
   }
   // If none of the above attempts succeed, accept the default value.
 
   TEUCHOS_TEST_FOR_EXCEPTION(
-    fillLevel <= 0, std::runtime_error,
+    fillLevel < 1.0, std::runtime_error,
```

Comment:
Note to other readers: this is the one way in which this change could break backwards compatibility.  Users might have set fill level to zero before, expecting that ILUT's fill level behaves like the "k" in ILU(k).  ILUT's fill level does _not_ behave like the "k" in ILU(k).  I am in favor of stopping users from doing this wrong thing, but this could possibly mean that users get throws when they didn't before.